### PR TITLE
feat: new component TimeSelectInput

### PR DESCRIPTION
**Pull request by rafaelmg83**
fix: #1490 

## Changes proposed in this PR:
- New Component TimeSelectInput

- [ X ] I have followed (at least) the [PR section of the contributing guide](https://github.com/nexxtway/react-rainbow/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

@nexxtway/react-rainbow

### CODE REVIEW

File: src/components/TimeSelectInput/index.js [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+        let normalizedValue;
+
+        if (isNumber(value)) {
+            this.value = value;
+            if (Number(value) > 23 || this.isUpOrDownKeyPressed || this.hasPropValue) {
+                const newTypedValue = getSingleNewTypedValue(hour, value);
+                normalizedValue = normalizeHour(newTypedValue, hour24);
+                this.setState({
+                    hour: normalizedValue,
+                });
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/index.js:118]: call handleChangeTime({
hour: normalizedValue,
});

File: src/components/TimeSelectInput/index.js [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+                const newTypedValue = getSingleNewTypedValue(minutes, value);
+                normalizedValue = normalizeMinutes(newTypedValue);
+                this.setState({
+                    minutes: normalizedValue,
+                });
+            } else {
+                normalizedValue = normalizeMinutes(value);
+                this.setState({
+                    minutes: normalizedValue,
+                });
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/index.js:189]: handleChangeTime({
minutes: normalizedValue,
});

File: src/components/TimeSelectInput/index.js [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+        const { hour24 } = this.props;
+        let normalizedValue;
+
+        if (isNumber(value)) {
+            if (Number(value) > 60 || this.isUpOrDownKeyPressed) {
+                const newTypedValue = getSingleNewTypedValue(minutes, value);
+                normalizedValue = normalizeMinutes(newTypedValue);
+                this.setState({
+                    minutes: normalizedValue,
+                });
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/index.js:184]: handleChangeTime({
minutes: normalizedValue,
});

File: src/components/TimeSelectInput/index.js [hunk 4]
```diff
@@ -0,0 +1,509 @@
...
+            minutes: '',
+        });
+    }
+
+    handleAmPmChange(value) {
+        this.setState({
+            inputFocusedIndex: -1,
+            ampm: value,
+        });
+        setTimeout(() => this.handleChangeTime(), 0);
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/index.js:217]: handleChangeTime({
inputFocusedIndex: -1,
            ampm: value,
});

File: src/components/TimeSelectInput/index.js [hunk 5]
```diff
@@ -0,0 +1,509 @@
...
+            currentInputFocused.current.focus();
+        }
+    }
+
+    resetState() {
+        const { inputFocusedIndex } = this.state;
+        if (inputFocusedIndex === 0) {
+            this.setState({
+                hour: '',
+            });
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/index.js:312]: handleChangeTime({
hour: '',
});

File: src/components/TimeSelectInput/index.js [hunk 6]
```diff
@@ -0,0 +1,509 @@
...
+            this.prevHour = '';
+        }
+        if (inputFocusedIndex === 1) {
+            this.setState({
+                minutes: '',
+            });
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/index.js:318]: handleChangeTime({
minutes: '',
});

File: src/components/TimeSelectInput/index.js [hunk 7]
```diff
@@ -0,0 +1,509 @@
...
+            this.prevMinutes = '';
+        }
+        setTimeout(() => this.handleChangeTime(), 0);
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/index.js:321]: delete this call

File: src/components/TimeSelectInput/index.js [hunk 8]
```diff
@@ -0,0 +1,509 @@
...
+    }
+
+    incrementHour() {
+        const { hour } = this.state;
+        const { hour24 } = this.props;
+        const hourValue = hour || this.prevHour;
+        this.setState({
+            hour: normalizeHour(getNextHour(hourValue, hour24), hour24),
+        });
+        setTimeout(() => this.handleChangeTime(), 0);
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/index.js:345]: handleChangeTime({
hour: normalizeHour(getNextHour(hourValue, hour24), hour24),
});

File: src/components/TimeSelectInput/index.js [hunk 9]
```diff
@@ -0,0 +1,509 @@
...
+    }
+
+    decrementHour() {
+        const { hour } = this.state;
+        const { hour24 } = this.props;
+        const hourValue = hour || this.prevHour;
+        this.setState({
+            hour: normalizeHour(getPrevHour(hourValue, hour24), hour24),
+        });
+        setTimeout(() => this.handleChangeTime(), 0);
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/index.js:355]: same before

File: src/components/TimeSelectInput/index.js [hunk 10]
```diff
@@ -0,0 +1,509 @@
...
+    }
+
+    incrementMinutes() {
+        const { minutes } = this.state;
+        const minutesValue = minutes || this.prevMinutes;
+        this.setState({
+            minutes: normalizeMinutes(getNextMinute(minutesValue)),
+        });
+        setTimeout(() => this.handleChangeTime(), 0);
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/index.js:364]: same before

File: src/components/TimeSelectInput/index.js [hunk 11]
```diff
@@ -0,0 +1,509 @@
...
+    }
+
+    decrementMinutes() {
+        const { minutes } = this.state;
+        const minutesValue = minutes || this.prevMinutes;
+        this.setState({
+            minutes: normalizeMinutes(getPrevMinute(minutesValue)),
+        });
+        setTimeout(() => this.handleChangeTime(), 0);
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/index.js:373]: same before

File: src/components/TimeSelectInput/readme.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+```js
+import React from 'react';
+import { TimeSelectInput } from 'react-rainbow-components';
+
+const containerStyles = {
+    display: "flex",
+    justifyContent: "center",
+};
+<div style={containerStyles}>
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/readme.md:11]: use styled-component for Container

File: src/components/TimeSelectInput/readme.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+```js
+import React from 'react';
+import { TimeSelectInput } from 'react-rainbow-components';
+
+const containerStyles = {
+    display: "flex",
+    justifyContent: "center",
+};
+
+<div style={containerStyles}>
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/readme.md:29]: same before

File: src/components/TimeSelectInput/readme.md [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/readme.md:48]: same before

File: src/components/TimeSelectInput/readme.md [hunk 15]
```diff
@@ -3,52 +3,55 @@
...
-</div>
+</StyledContainer>
 ```
 
 ##### TimeSelectInput with initial value:
 
 ```js
 import React from 'react';
 import { TimeSelectInput } from 'react-rainbow-components';
+import styled from 'styled-components';
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/readme.md:24]: Same before

File: src/components/TimeSelectInput/readme.md [hunk 16]
```diff
@@ -3,52 +3,55 @@
...
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/readme.md:6]: External import before internal import always

File: src/components/TimeSelectInput/readme.md [hunk 17]
```diff
@@ -3,52 +3,55 @@
...
-</div>
+</StyledContainer>
 ```
 
 ##### TimeSelectInput with 24 hour format:
 
 ```js
 import React from 'react';
 import { TimeSelectInput } from 'react-rainbow-components';
+import styled from 'styled-components';
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/readme.md:44]: Same before

File: src/components/TimeSelectInput/index.js [hunk 18]
```diff
@@ -0,0 +1,503 @@
...
+        const { hour24 } = this.props;
+        let normalizedValue;
+
+        if (isNumber(value)) {
+            if (Number(value) > 60 || this.isUpOrDownKeyPressed) {
+                const newTypedValue = getSingleNewTypedValue(minutes, value);
+                normalizedValue = normalizeMinutes(newTypedValue);
+                this.setState({
+                    minutes: normalizedValue,
+                });
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/index.js:185]: I have here you shouldn't call handleChangeTime (newState) to call onChange ()

File: src/components/TimeSelectInput/readme.md [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+
+const StyledContainer = styled.div`
+    display: flex;
+    justify-content: center;
+`;
+<StyledContainer>
+    <TimeSelectInput
+        id="time-select-input-1"
+    />;
+</StyledContainer>
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/readme.md:16]: All Inputs use the "value" and "onChange ()" pattern. To do the examples it had to be like this.

`const TimeSelectInputExample = () => {
    const [time, setTime] = useState('08:00 AM');
    return (
        <TimeSelectInput
            id="time-select-input-2"
            value="08:00 AM"
            onChange={setTime}
        />
    );
}
<StyledContainer>
    <TimeSelectInputExample />
</StyledContainer>
`

File: src/components/TimeSelectInput/readme.md [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+    display: flex;
+    justify-content: center;
+`;
+
+<StyledContainer>
+    <TimeSelectInput
+        id="time-select-input-2"
+        value="08:00 AM"
+    />;
+</StyledContainer>
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/readme.md:36]: same before

File: src/components/TimeSelectInput/readme.md [hunk 21]
```diff
@@ -0,0 +1,57 @@
...
+    display: flex;
+    justify-content: center;
+`;
+
+<StyledContainer>
+    <TimeSelectInput
+        id="time-select-input-2"
+        hour24={true}
+    />;
+</StyledContainer>
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/readme.md:56]: same before

File: src/components/TimeSelectInput/index.js [hunk 22]
```diff
@@ -0,0 +1,503 @@
...
+            minutes: currentMinutes,
+            ampm: currentAmPm,
+        });
+        this.setState(newState);
+
+        if (
+            (currentHour && currentMinutes && ampm && !hour24) ||
+            (currentHour && currentMinutes && hour24)
+        ) {
+            onChange(time);
```
Comment (yvmunayev) [inline src/components/TimeSelectInput/index.js:385]: @rafaelmg83  The behavior of all our Input components is, that when you change the value you have to call the onChange method. You have to check that behavior well in this component, it is working fine.
I already told you to change the examples to reflect that behavior, so you can try adding a console.log (value) in the onChange and check that every time you change the date in the component it makes you console.log.

Comment (rafaelmg83) [reply src/components/TimeSelectInput/index.js:385]: I don't know what you mean exactly, but the onChange event is called every time that any part of hour, minutes or am/pm changes except the value is incomplete where it has no logic to call it. Of course I tried the console.log test before push. Maybe you are seeing something that I am not, but I think that behavior is fine as it is.

File: src/components/TimeSelectInput/readme.md [hunk 23]
```diff
@@ -0,0 +1,78 @@
...
+const StyledContainer = styled.div`
+    display: flex;
+    justify-content: center;
+`;
+const TimeSelectInputExample = () => {
+    const [time, setTime] = useState(null); 
+    return ( 
+        <TimeSelectInput 
+            id="time-select-input-2" 
+            value=""
```
Comment (LeandroTorresSicilia) [inline src/components/TimeSelectInput/readme.md:17]: since this component is an inputable it must be controlled, then the source of truth is the value prop, then if you don't pass the time state here as the value it should not work

File: src/components/TimeSelectInput/index.d.ts [hunk 24]
```diff
@@ -0,0 +1,12 @@
+import { ComponentType } from 'react';
+import { BaseProps } from '../types';
+
+export interface TimeSelectInputProps extends BaseProps {
+    value?: string;
+    onChange?: (time: string) => void;
```
[the remark this answers is not in the document]
Comment (rafaelmg83) [reply src/components/TimeSelectInput/index.d.ts:6]: @yvmunayev I fixed the first error already, but, I don't understand what you mean with the second one, if it is the case you try to enter a number higher (6-9) than 5 [minutes: 0-59] it assumes that is the entire minute and focus the next input (am/pm). By ex. if you enter 7 the minutes will be: 07. Makes sense?

[the remark this answers is not in the document]
Comment (LeandroTorresSicilia) [reply src/components/TimeSelectInput/index.d.ts:6]: @yvmunayev We discussed this and we think it should be a string with the time as a string formatted and ready to store in a database or whatever place the user can use, right now the TimePicker send it in this way too (as a formatted string). But we can discuss it again to see your points.

File: src/components/TimeSelectInput/helpers/index.js [hunk 25]
```diff
@@ -0,0 +1,39 @@
+import get12HourTime from './get12HourTime';
```
Comment (LeandroTorresSicilia) [inline src/components/TimeSelectInput/helpers/index.js:1]: here you can:
```
export { default as get12HourTime } from './get12HourTime';
export { default as get24HourTime } from './get24HourTime';
...
```
and then it gets shorter importing and exporting in only one line each one

File: src/components/TimeSelectInput/icons/index.js [hunk 26]
```diff
@@ -0,0 +1,4 @@
+import DownArrow from './downArrow';
+import UpArrow from './upArrow';
+
+export { DownArrow, UpArrow };
```
Comment (LeandroTorresSicilia) [inline src/components/TimeSelectInput/icons/index.js:4]: same ^^

**Comment by LeandroTorresSicilia** [comment]
- when the hour input has a value and press delete key it does not clean its value, we should add test to this behavior

File: src/components/TimeSelectInput/index.js [hunk 27]
```diff
@@ -0,0 +1,505 @@
...
+            this.handleLeftKeyPressed();
+        }
+        if (keyCode === UP_KEY) {
+            this.incrementHandler();
+        }
+        if (keyCode === DOWN_KEY) {
+            this.decrementHandler();
+        }
+        if (keyCode === DELETE_KEY) {
+            this.resetState();
```
Comment (LeandroTorresSicilia) [inline src/components/TimeSelectInput/index.js:238]: this is not working properly since the hour input value does not reset, we should add test to this behavior

File: src/components/TimeSelectInput/index.js [hunk 28]
```diff
@@ -0,0 +1,505 @@
...
+        const currentInputFocused = this.inputsMap[inputFocusedIndex];
+        if (currentInputFocused) {
+            currentInputFocused.current.focus();
+        }
+    }
+
+    resetState() {
+        const { inputFocusedIndex } = this.state;
+        if (inputFocusedIndex === 0) {
+            this.handleChangeTime()({
```
Comment (LeandroTorresSicilia) [inline src/components/TimeSelectInput/index.js:307]: sure this is the problem resetting the hour value you are calling it twice and the first time without arguments:
```this.handleChangeTime()({ ... })```

File: src/components/TimeSelectInput/styled/index.js [hunk 29]
```diff
@@ -0,0 +1,301 @@
...
+export {
+    StyledContainer,
+    StyledDots,
+    StyledDownArrow,
+    StyledInputHidden,
+    StyledOptionLabel,
+    StyledSelectContent,
+    StyledSelectValue,
+    StyledUpArrow,
+    StyledVerticalButtonsContainer,
```
Comment (LeandroTorresSicilia) [inline src/components/TimeSelectInput/styled/index.js:300]: you can export directly in each const

**Comment by LeandroTorresSicilia** [comment]
We also need to add test to index, currently we have test of this in `src/components/TimePicker/__test__/timeSelect.spec.js` and also add integration tests but you can wait to have the correct behavior to add these tests in order to make the right changes.

File: src/components/TimeSelectInput/helpers/normalizeValue.js [hunk 30]
```diff
@@ -0,0 +1,56 @@
...
+    if (!hour24 && Number(hour) < 12) {
+        if (!ampm) ampm = 'AM';
+        if (hour === '00') hour = 12;
+    }
+
+    return {
+        hour: normalizeHour(hour, hour24),
+        minutes: normalizeMinutes(minutes, hour24),
+        ampm,
+    };
```
Comment (LeandroTorresSicilia) [inline src/components/TimeSelectInput/helpers/normalizeValue.js:55]: we also need to add test to this function

**Comment by TahimiLeonBravo** [comment]
@rafaelmg83, when navigating to the component page all the TimeSelectInput examples, have focused the hour's element 
Screen Shot 2020-05-04 at 9 45 16 AM

File: src/components/TimeSelectInput/index.d.ts (hunk 24 above)
[the remark this answers is not in the document]
Comment (yvmunayev) [reply src/components/TimeSelectInput/index.d.ts:6]: @rafaelmg83  try putting 12:12 on the keyboard several times

[the remark this answers is not in the document]
Comment (rafaelmg83) [reply src/components/TimeSelectInput/index.d.ts:6]: > @rafaelmg83 try putting 12:12 on the keyboard several times

fixed! Thanks

**Comment by rafaelmg83** [comment]
> check that you have some lint problems

fixed! Thanks.

File: src/components/WeeklyCalendar/readme.md [hunk 31]
```diff
@@ -96,7 +96,7 @@ const initialState = {
     currentWeek: new Date(),
 };
 
-<div className="rainbow-m-around_large">
+<div className="rainbow-m-around_large ">
```
Comment (yvmunayev) [inline src/components/WeeklyCalendar/readme.md:99]: check this

**Comment by LeandroTorresSicilia** [comment]
We need to be able to set the focus programmatically like when de component mount the first time:
```
  const ref = useRef();
  	useEffect(() => {
    	ref.current.focus()
    }, []);
    return (
        <TimeSelectInput
            id="time-select-input-1"
            value={time} 
            onChange={setTime}
            label="TimeSelectInput Label"
            ref={ref}
        />
```

**Comment by LeandroTorresSicilia** [comment]
This should not happen, if value is `08:00 A` then the component should render the placeholder `--:-- --`
Screen Shot 2020-05-23 at 9 16 34 AM

**Comment by LeandroTorresSicilia** [comment]
@TahimiLeonBravo check disabled and readOnly styles since there is no visual difference between them

File: src/components/TimeSelectInput/index.d.ts [hunk 32]
```diff
@@ -0,0 +1,23 @@
...
+    hideLabel?: boolean;
+    required?: boolean;
+    bottomHelpText?: ReactNode;
+    error?: ReactNode;
+    readOnly?: boolean;
+    disabled?: boolean;
+    tabIndex?: number | string;
+    onClick?: (event: MouseEvent<HTMLInputElement>) => void;
+    onFocus?: (value: string) => void;
+    onBlur?: (value: string) => void;
```
Comment (LeandroTorresSicilia) [inline src/components/TimeSelectInput/index.d.ts:17]: There are inconsistencies here, currently, blur sends a string and it is fired only once like if it was only one input (good until here), but initially blur without any value it sends `null` instead of an empty string.
In the case of the focus, it behaves completely different, it fires an event object instead of the string and it fires every time for each input.
Let me check but I think we should send on both a string empty when no value or incomplete or the value when correct, and treat the input as only one, it means it should not send focus for each input

File: src/components/TimeSelectInput/index.js [hunk 33]
```diff
@@ -94,6 +100,7 @@ export default class TimeSelectInput extends Component {
 
     handleClickOutside() {
         const { inputFocusedIndex } = this.state;
+        this.handleBlur();
```
Comment (LeandroTorresSicilia) [inline src/components/TimeSelectInput/index.js:103]: this is causing the blur to get called every time for each input, this is not a correct behaviour since only focused element should be able to blur

File: src/components/TimeSelectInput/index.js [hunk 34]
```diff
@@ -0,0 +1,633 @@
...
+        this.containerRef.current.focus();
+    }
+
+    /**
+     * Sets click on the element.
+     * @public
+     */
+    click() {
+        this.containerRef.current.click();
+    }
```
Comment (LeandroTorresSicilia) [inline src/components/TimeSelectInput/index.js:426]: we need to add blur method too, like all other input components

File: src/components/TimeSelectInput/index.js [hunk 35]
```diff
@@ -0,0 +1,633 @@
...
+                        onChange={this.handleChangeHour}
+                        onFocus={this.handleFocusHour}
+                        onBlur={this.handleBlurHour}
+                        inputMode="numeric"
+                        isFocused={inputFocusedIndex === 0}
+                        pattern="\d*"
+                        ref={this.hourInputRef}
+                        disabled={disabled}
+                        readOnly={readOnly}
+                        tabIndex={tabIndex || null}
```
Comment (LeandroTorresSicilia) [inline src/components/TimeSelectInput/index.js:499]: There is a reason to use null?
If not I think we should simply do:
```tabIndex={tabIndex}```

File: src/components/TimeSelectInput/index.js [hunk 36]
```diff
@@ -408,11 +464,20 @@ export default class TimeSelectInput extends Component {
...
 
         return (
-            <article className={className} style={style} id={id}>
+            <StyledContainer className={className} style={style} id={id}>
+                <RenderIf isTrue={!!label}>
+                    <StyledLabel
+                        label={label}
+                        hideLabel={hideLabel}
+                        required={required}
+                        inputId="timeselectinput-label"
```
Comment (LeandroTorresSicilia) [inline src/components/TimeSelectInput/index.js:473]: we need to research here what is the best approach for accessibility since this is for associate the input with the label, we can't simply pass a fixed text here, also normally this value is generated and unique since it can't be used by another element in the DOM

File: src/components/TimeSelectInput/readme.md [hunk 37]
```diff
@@ -69,6 +71,92 @@ const TimeSelectInputExample = () => {
...
+
+const StyledContainer = styled.div`
+    display: flex;
+    justify-content: center;
+`;
+const TimeSelectInputExample = () => {
+    const [time, setTime] = useState(null); 
+    return ( 
+        <TimeSelectInput 
+            id="time-select-input-5"
```
Comment (LeandroTorresSicilia) [inline src/components/TimeSelectInput/readme.md]: ids must be unique, I see you are using `time-select-input-5` in each one

**Comment by rafaelmg83** [comment]
> This should not happen, if value is `08:00 A` then the component should render the placeholder `--:-- --`
> Screen Shot 2020-05-23 at 9 16 34 AM

Fixed!

**Comment by rafaelmg83** [comment]
> We need to be able to set the focus programmatically like when de component mount the first time:
> 
> ```
>   const ref = useRef();
>     useEffect(() => {
>     	ref.current.focus()
>     }, []);
>     return (
>         <TimeSelectInput
>             id="time-select-input-1"
>             value={time} 
>             onChange={setTime}
>             label="TimeSelectInput Label"
>             ref={ref}
>         />
> ```

Fixed!

**Comment by rafaelmg83** [comment]
> @TahimiLeonBravo check disabled and readOnly styles since there is no visual difference between them

Fixed!

**Comment by rafaelmg83** [comment]
> @rafaelmg83, when navigating to the component page all the TimeSelectInput examples, have focused the hour's element
> Screen Shot 2020-05-04 at 9 45 16 AM

Fixed!

File: src/components/TimeSelectInput/index.js [hunk 38]
```diff
@@ -596,6 +608,8 @@ export default class TimeSelectInput extends Component {
     }
 }
 
+export default withForwardedRef(TimeSelectInput);
```
Comment (LeandroTorresSicilia) [inline src/components/TimeSelectInput/index.js:612]: we need to export this component withReduxForm HOC

File: src/components/TimeSelectInput/helpers/normalizeValue.js [hunk 39]
```diff
@@ -16,7 +16,9 @@ export default function normalizeValue(value, hour24) {
         ampm: '',
     };
 
-    if (!value) return invalidValue;
+    if (!value || hour24 === undefined || hour24 === null || typeof hour24 !== 'boolean') {
```
Comment (LeandroTorresSicilia) [inline src/components/TimeSelectInput/helpers/normalizeValue.js:19]: only checking for ```typeof hour24 !== 'boolean'``` is sufficent

File: src/components/TimeSelectInput/helpers/__test__/normalizeValue.spec.js [hunk 40]
```diff
@@ -0,0 +1,137 @@
...
+        return '12';
+    }
+    if (+value < 10) {
+        return `0${+value}`;
+    }
+    if (+value > 12 && +value < 20 && !hour24) {
+        return `0${+value - 12}`;
+    }
+    return String(+value);
+});
```
Comment (LeandroTorresSicilia) [inline src/components/TimeSelectInput/helpers/__test__/normalizeValue.spec.js:71]: we should not reimplement helpers in its mocks, since it is difficult to read and understand, also how you guaranties the right value returned, if we use this complex implementations in the mocks the we would need to test the mock itself, instead what you can do is to return a simple value like an empty string and then in each test where a specific value is needed, you can do:
```normalizeHour.mockReturnValue('12')```

**Comment by LeandroTorresSicilia** [comment]
@TahimiLeonBravo check this styles, when readonly it has different colors
Screen Shot 2020-05-30 at 9 17 40 AM

**Comment by rafaelmg83** [comment]
> @TahimiLeonBravo check this styles, when readonly it has different colors
> Screen Shot 2020-05-30 at 9 17 40 AM

Fixed!
